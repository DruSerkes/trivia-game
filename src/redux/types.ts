export const ADD_QUESTIONS = 'ADD_QUESTIONS';
export const REMOVE_QUESTIONS = 'REMOVE_QUESTIONS';
export const QUESTIONS_FAIL = 'QUESTIONS_FAIL';

interface Question {
	question: string;
	correct_answer: string;
	incorrect_answers: string[];
	[propName: string]: any;
}

interface AddQuestions {
	type: typeof ADD_QUESTIONS;
	payload: Question[];
}

interface QuestionsFail {
	type: typeof QUESTIONS_FAIL;
}

interface RemoveQuestions {
	type: typeof REMOVE_QUESTIONS;
}

export type QuestionType = Question;
export type QuestionsDispatchTypes = AddQuestions | QuestionsFail | RemoveQuestions;
