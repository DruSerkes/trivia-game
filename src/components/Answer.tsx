import * as React from 'react';
import { ListItem } from '@material-ui/core'
import { showIfCorrect } from '../helpers/addClass'

type AnswerProps = {
    answer: string;
    answerQuestion: (answer: string) => void;
    correct?: boolean | null;
}

export function Answer({ answer, answerQuestion, correct }: AnswerProps) {
    const renderHTML = () => (<span className="AnswerText" dangerouslySetInnerHTML={{ __html: answer }} />)
    const handleClick = ({ nativeEvent: e }: React.MouseEvent) => {
        showIfCorrect(e, correct);
        setTimeout(() => answerQuestion(answer), 600);
    }
    return (
        <ListItem id={answer} className="Answer" divider onClick={handleClick}>{renderHTML()}</ListItem>
    );
}