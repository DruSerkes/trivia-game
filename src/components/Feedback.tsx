import * as React from 'react';



export const Feedback = () => (< p className="Feedback" data-testid="Feedback" />)