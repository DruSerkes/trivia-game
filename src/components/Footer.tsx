import React from "react";

export const Footer = () => {
  const year = new Date().getFullYear();
  return (
    <footer className="Footer">
      <p>
        <a
          data-testid="footer"
          href="https://github.com/druserkes"
          target="_blank"
          rel="noopener noreferrer"
        >
          Dru Serkes &copy;{year}
        </a>
      </p>
    </footer>
  );
};
