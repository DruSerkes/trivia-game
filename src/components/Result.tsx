import * as React from 'react';
import { Box, Typography } from '@material-ui/core'
import { renderColor } from '../helpers/renderColor'

type ResultTypes = {
    numCorrect: number;
    numAnswered: number;
}

export const Result = ({ numCorrect, numAnswered }: ResultTypes) => {
    return (
        <Box className="Result">
            <Typography variant="h3">Game Over!</Typography>
            <Typography variant="body1" style={{ color: renderColor(numCorrect) }}>You answered {numCorrect} out of {numAnswered} correct</Typography>
        </Box>
    )
}