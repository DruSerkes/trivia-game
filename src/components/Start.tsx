import * as React from 'react';
import { useState } from 'react';
import { Box, Button, TextField, FormControl, RadioGroup, FormControlLabel, Radio, FormLabel } from '@material-ui/core'

type StartProps = {
    startGame: (amount: string, difficulty: string, type: string) => void
}

export const Start = ({ startGame }: StartProps) => {
    const [amount, setAmount] = useState<string>('10');
    const [difficulty, setDifficulty] = useState<string>('');
    const [type, setType] = useState<string>('multiple');

    const handleChangeAmount = (e: React.ChangeEvent<HTMLInputElement>): void => setAmount(e.target.value);
    const handleChangeDifficulty = (e: React.ChangeEvent<HTMLInputElement>): void => setDifficulty(e.target.value);
    const handleChangeType = (e: React.ChangeEvent<HTMLInputElement>): void => setType(e.target.value);
    const handleSubmit = () => startGame(amount, difficulty, type);

    return (
        <Box className="Start">
            <TextField label="Number of Questions" type="number" value={amount} onChange={handleChangeAmount} />
            <FormControl component="fieldset">
                <FormLabel component="legend">Difficulty</FormLabel>
                <RadioGroup row aria-label="difficulty" name="difficulty" value={difficulty} onChange={handleChangeDifficulty}>
                    <FormControlLabel value="easy" control={<Radio />} label="Easy" labelPlacement="end" />
                    <FormControlLabel value="medium" control={<Radio />} label="Medium" labelPlacement="end" />
                    <FormControlLabel value="hard" control={<Radio />} label="Hard" labelPlacement="end" />
                    <FormControlLabel value="" control={<Radio />} label="Surprise Me!" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <FormControl component="fieldset">
                <FormLabel component="legend">Type</FormLabel>
                <RadioGroup row aria-label="type" name="type" value={type} onChange={handleChangeType}>
                    <FormControlLabel value="multiple" control={<Radio />} label="Multiple Choice" labelPlacement="end" />
                    <FormControlLabel value="boolean" control={<Radio />} label="True/False" labelPlacement="end" />
                </RadioGroup>
            </FormControl>
            <Button variant="contained" fullWidth color="primary" onClick={handleSubmit}>Start New Game</Button>
        </Box >
    )
}