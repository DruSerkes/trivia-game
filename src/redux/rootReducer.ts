import { questionsReducer } from './questionsReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({ questions: questionsReducer });

export default rootReducer;
