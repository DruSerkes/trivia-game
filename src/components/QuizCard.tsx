import * as React from 'react';
import { QuestionType } from '../redux/types';

type QuizCardProps = {
    question: QuestionType | null;
    answerQuestion: (answer: string) => void
}


export const QuizCard = ({ question, answerQuestion }: QuizCardProps) => {
    // Render the 3 incorrect answers and the correct answer
    // Do so with each being a separate Answer component
    // Answer component should accept props for question and isCorrect (optional bool)
    // When Answer is clicked, it should fire off answerQuestion with the answer passed in


    return (
        <div className="QuizCard">
            <h4>{question?.question}</h4>
            {/* RENDER 4 ANSWERS HERE */}
        </div>
    )
}