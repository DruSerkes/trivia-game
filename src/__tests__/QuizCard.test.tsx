import React from 'react';
import { render } from '@testing-library/react';
import thunk from 'redux-thunk';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../redux/rootReducer'
import { composeWithDevTools } from 'redux-devtools-extension';
import { QuizCard } from '../components/QuizCard'
import { QuestionType } from '../redux/types';

const store = createStore(rootReducer, composeWithDevTools(applyMiddleware(thunk)));
const question: QuestionType = {
    question: 'testing',
    correct_answer: 'true',
    incorrect_answers: ['false', 'false too', 'also false']
}
test('QuizCard Matches Snapshot ', () => {
    const { asFragment } = render(
        <Provider store={store}>
            <QuizCard question={question} answerQuestion={() => { }} />
        </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});