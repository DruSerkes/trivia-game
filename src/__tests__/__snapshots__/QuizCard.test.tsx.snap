// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCard Matches Snapshot  1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1 QuizCard"
    data-testid="QuizTest"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      testing
    </h5>
    <ul
      class="MuiList-root QuizCardList MuiList-padding"
    >
      <li
        class="MuiListItem-root Answer MuiListItem-gutters MuiListItem-divider"
      >
        <span
          class="AnswerText"
        >
          true
        </span>
      </li>
      <li
        class="MuiListItem-root Answer MuiListItem-gutters MuiListItem-divider"
      >
        <span
          class="AnswerText"
        >
          false
        </span>
      </li>
      <li
        class="MuiListItem-root Answer MuiListItem-gutters MuiListItem-divider"
      >
        <span
          class="AnswerText"
        >
          falsey
        </span>
      </li>
      <li
        class="MuiListItem-root Answer MuiListItem-gutters MuiListItem-divider"
      >
        <span
          class="AnswerText"
        >
          false-ish
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
